tests/validation/tests/v3_api/common.py

```diff
@@ -1256,20 +1266,28 @@ def check_connectivity_between_workload_pods(p_client, workload):
 def check_connectivity_between_pods(pod1, pod2, allow_connectivity=True):
     pod_ip = pod2.status.podIp
 
-    cmd = "ping -c 1 -W 1 " + pod_ip
     if is_windows():
         cmd = 'ping -w 1 -n 1 {0}'.format(pod_ip)
+    elif HARDENED_CLUSTER:
+        cmd = 'curl -I {}:{}'.format(pod_ip, TEST_IMAGE_PORT)
+    else:
+        cmd = "ping -c 1 -W 1 " + pod_ip
 
     response = kubectl_pod_exec(pod1, cmd)
-    assert pod_ip in str(response)
+    if not HARDENED_CLUSTER:
```

Comment:
Do we need checks for when HARDENED_CLUSTER=True ?

Reply:
We are just relying on the below `allow_connectivity` (or disallow) checks in this case. This is simply because `ping` returns the IP that a domain name resolves to but `curl` does not, and there isn't a similar check in the case of `curl` so I'm just skipping it.

Reply:
okay.